### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by rai-nhdaly**
Updated the README/docstring to reflect the recent changes (https://github.com/RelationalAI-oss/Rematch.jl/pull/9 and https://github.com/RelationalAI-oss/Rematch.jl/pull/11).

And expanded the intro paragraph a bit to talk more about what this Package does.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,11 +61,23 @@ Stacktrace:
...
 
+### Assignment Syntax
 ``` julia
 @match pattern = value
 ```
 
-If value matches pattern, binds variables and returns `value`. Otherwise, throws MatchFailure.
+If value matches pattern, binds variables and returns `value`. Otherwise, throws `MatchFailure`.
+
+After evaluation, any variables defined within `pattern` will be bound as new variables in the enclosing scope. e.g.:
```
Comment (mbravenboer) [inline README.md]: 'defined' is maybe confusing? perhaps 'used'? They're not necessarily new variables, right?

File: README.md [hunk 2]
```diff
@@ -70,13 +87,16 @@ If value matches pattern, binds variables and returns `value`. Otherwise, throws
 end
 ```
 
-Returns `result` for the first matching pattern. If there are no matching patterns, throws MatchFailure.
+Returns `result` for the first matching pattern. If there are no matching patterns, throws `MatchFailure`.
+
+Note that unlike the _assignment syntax_, this does not create any variable bindings outside the match macro.
```
Comment (mbravenboer) [inline README.md]: Is this true? I thought that it would re-assign to existing variables?

File: README.md [hunk 3]
```diff
@@ -86,16 +106,21 @@ Returns `result` for the first matching pattern. If there are no matching patter
 * `x && y` matches values which match both `x` and `y`
 * `x where condition` matches only if `condition` is true (`condition` may use any variables that occur earlier in the pattern eg `(x, y, z where x + y > z)`)
 * Anything else is treated as a constant and tested for equality
+* Expressions can be interpolated in as constants via standard interpolation syntax `$()`
```
Comment (mbravenboer) [inline README.md]: Maybe add a small example here as with the other cases? For example,  $(x) would already be sufficient.

File: README.md (hunk 2 above)
Comment (rai-nhdaly) [reply README.md]: I think only the assignment syntax does this right now.

See, for example, these tests:
https://github.com/RelationalAI-oss/Rematch.jl/blob/master/test/rematch.jl#L33-L39
vs these:
https://github.com/RelationalAI-oss/Rematch.jl/blob/master/test/rematch.jl#L310-L312

Comment (mbravenboer) [reply README.md]: Ah thanks for the clarification

File: README.md (hunk 1 above)
Comment (rai-nhdaly) [reply README.md]: Updated in latest commit. thanks!

File: README.md (hunk 3 above)
Comment (rai-nhdaly) [reply README.md]: Yeah, I agree, I think just that -- `$(x)` -- is sufficient. thanks! Updated in latest commit.

**Comment by rai-nhdaly** [comment]
Merging. Thanks!

**Comment by rai-nhdaly** [comment]
(Partially fixes https://github.com/RelationalAI-oss/Rematch.jl/issues/14)

**Pull request merged**